ansible_collections/arista/avd/plugins/plugin_utils/eos_designs_facts/uplinks.py

```diff
@@ -184,148 +184,239 @@ def uplinks(self: EosDesignsFacts) -> list:
+                # Invalid length of input variables. Skipping
+                continue
 
-                if self.shared_utils.uplink_structured_config is not None:
-                    uplink["structured_config"] = self.shared_utils.uplink_structured_config
+            uplink_switch = uplink_switches[uplink_index]
+            if uplink_switch is None or uplink_switch not in self.shared_utils.all_fabric_devices:
+                # Invalid uplink_switch. Skipping.
+                continue
 
-                uplinks.append(uplink)
+            uplink_switch_facts: EosDesignsFacts = self.shared_utils.get_peer_facts(uplink_switch, required=True)
+            uplink_peer_interface = uplink_switch_interfaces[uplink_index]
+            uplink = {
+                "interface": uplink_interface,
+                "peer": uplink_switch,
+                "peer_interface": uplink_peer_interface,
+                "peer_type": uplink_switch_facts.type,
+                "peer_is_deployed": uplink_switch_facts.is_deployed,
+                "peer_bgp_as": uplink_switch_facts.bgp_as,
+                "type": "underlay_p2p_vrfs",
+            }
+            if self.shared_utils.uplink_interface_speed is not None:
+                uplink["speed"] = self.shared_utils.uplink_interface_speed
+
+            if self.shared_utils.uplink_bfd:
+                uplink["bfd"] = True
+
+            if self.shared_utils.uplink_switch_interface_speed is not None:
+                uplink["peer_speed"] = self.shared_utils.uplink_switch_interface_speed
+
+            # if self.shared_utils.uplink_ptp is not None:
+            #    uplink["ptp"] = self.shared_utils.uplink_ptp
+            # elif self.shared_utils.ptp_enabled:
+            #    uplink["ptp"] = {"enable": True}
+
+            # if self.shared_utils.uplink_macsec is not None:
+            #    uplink["mac_security"] = self.shared_utils.uplink_macsec
+
+            # if self.shared_utils.underlay_multicast is True and uplink_switch_facts.shared_utils.underlay_multicast is True:
+            #    uplink["underlay_multicast"] = True
+
+            # if self.shared_utils.link_tracking_groups is not None:
+            #    uplink["link_tracking_groups"] = []
+            #    for lt_group in self.shared_utils.link_tracking_groups:
+            #        uplink["link_tracking_groups"].append({"name": lt_group["name"], "direction": "upstream"})
+
+            uplink["subinterfaces"] = self._uplink_sub_interfaces(
+                uplink_interface, uplink_peer_interface, uplink_index, self.shared_utils.uplink_structured_config
+            )
+
+            uplinks.append(uplink)
         return uplinks
 
+    def _uplink_sub_interfaces(self: EosDesignsFacts, uplink_interface: str, uplink_peer_interface: str, uplink_index: int, struct_cfg: dict | None) -> list:
+        subinterfaces = []
+        for tenant in self.shared_utils.filtered_tenants:
```

Comment:
Somehow we need to check if the VRF is also available on the parent switch. 
We should scan the downstream switches for used VRFs similar to how we do it for VLANs. that way you will "attract" a VRF to the parent, even if it is only defined on the child.

Reply:
do we want to "attract: a VRF on the parent if it is only on the child? I think we said we wanted to not add it on the link rather.

Reply:
Right we decided to not follow the same logic as for VLANs. Instead we will only include VRFs on the uplinks if they are available on both ends of the link.